SU2_CFD/src/solvers/CHeatSolver.cpp

```diff
@@ -420,23 +352,23 @@ void CHeatSolver::Upwind_Residual(CGeometry *geometry, CSolver **solver_containe
 
   su2double *V_i, *V_j, Temp_i, Temp_i_Corrected, Temp_j, Temp_j_Corrected, **Gradient_i, **Gradient_j, Project_Grad_i, Project_Grad_j,
             **Temp_i_Grad, **Temp_j_Grad, Project_Temp_i_Grad, Project_Temp_j_Grad;
-  unsigned short iDim, iVar;
-  unsigned long iEdge, iPoint, jPoint;
-  bool flow = ((config->GetKind_Solver() == INC_NAVIER_STOKES)
-               || (config->GetKind_Solver() == INC_RANS)
-               || (config->GetKind_Solver() == DISC_ADJ_INC_NAVIER_STOKES)
-               || (config->GetKind_Solver() == DISC_ADJ_INC_RANS));
-  bool muscl = (config->GetMUSCL_Heat());
+
+  vector<su2double> Vector_i(nDim, 0.0);
+  vector<su2double> Vector_j(nDim, 0.0);
 
   if(flow) {
 
     nVarFlow = solver_container[FLOW_SOL]->GetnVar();
 
-    for (iEdge = 0; iEdge < geometry->GetnEdge(); iEdge++) {
+    /*--- Define some auxiliary vectors related to the primitive flow solution ---*/
+    vector<su2double> Primitive_Flow_i(nVarFlow, 0.0);
+    vector<su2double> Primitive_Flow_j(nVarFlow, 0.0);
```

Comment:
I get it that you want to clean this, but we might as well work towards a better scalar solver framework (as we have talked about), because all these things are going to be deleted...

Reply:
What do you mean with `all these things`?
- just these containers? - they are now no more class data so i wouldn't matter as it only lives in this very narrow scope. 
- If you mean the transported temperature scalar for incompressible flow alltogehter I fully agree. I didnt't want to open the discussion here but let's go for it : 

Can we delete the weakly coupled heat solver for incompressible flow in its entirety? From my incomplete perspective: yes

Or do you mean the whole heatsolver - with solid heat conduction as a special case of a (non-)transported scalar?

Reply:
Fair enough. I mean methods like UpwindResidual, ViscousResidual, SetTimeStep, DualTimeResidual, etc. etc. will have a common implementation across all scalar solvers.

Reply:
i wont go further in that matter in this PR though and I'd suggest to talk about this maybe briefly on Wednesday